src/tmpfiles/tmpfiles.c

```diff
@@ -2003,6 +2005,177 @@ static int glob_item_recursively(Item *i, fdaction_t action) {
         return r;
 }
 
+static int rm_if_wrong_type_safe(
+                mode_t mode,
+                int parent_fd,
+                const struct stat *parent_st /* Only used if follow is true. */,
+                const char *name,
+                bool follow) {
+        _cleanup_free_ char *parent_name = NULL;
+        struct stat st;
+        int r;
+        const char *e;
+
+        assert(name);
+        assert((mode & ~S_IFMT) == 0);
+        if (follow)
+                assert(parent_st);
+
+        e = strrchr(name, '/');
+        if (e)
+                return -EINVAL;
+
+        r = fstatat(parent_fd, name, &st, follow ? 0 : AT_SYMLINK_NOFOLLOW);
+        if (r < 0) {
```

Comment:
why not: 

```c
if (fstat() < 0) …
```

i.e. no need for the `r` here if you are not going to use this elsewhere